cli/azd/internal/tracing/attributes.go

```diff
@@ -78,6 +78,55 @@ func appendTo(v *valSynced, attr attribute.KeyValue) {
 	v.val.Store(newBaggage)
 }
 
+func appendToUnique(v *valSynced, attr attribute.KeyValue) {
+	v.mu.Lock()
+	defer v.mu.Unlock()
+
+	baggage := v.val.Load().(baggage.Baggage)
+	val, ok := baggage.Lookup(attr.Key)
+	if ok && val.Type() == attribute.STRINGSLICE {
+		switch attr.Value.Type() {
+		case attribute.STRING:
+			attrValue := attr.Value.AsString()
+			for _, v := range val.AsStringSlice() {
+				if v == attrValue { // already exists
+					return
+				}
+			}
+
+			attr = attr.Key.StringSlice(append(val.AsStringSlice(), attrValue))
+		case attribute.STRINGSLICE:
+			attrValues := attr.Value.AsStringSlice()
+			adds := make([]string, 0, len(attrValues))
+
+			for _, a := range attrValues {
+				found := false
+				for _, v := range val.AsStringSlice() {
+					if v == a {
```

Comment:
nit. Move the entire switch to a new function to reduce a little indentation

Reply:
`slices.Contains` might be helpful here.